samples/snippets/src/test/java/com/example/spanner/SpannerSampleIT.java

```diff
@@ -93,10 +97,45 @@ public static void setUp() throws Exception {
     dbId = DatabaseId.of(options.getProjectId(), instanceId, databaseId);
     // Delete stale test databases that have been created earlier by this test, but not deleted.
     deleteStaleTestDatabases(instanceId, baseDbId);
-    key = String.format("projects/%s/locations/%s/keyRings/%s/cryptoKeys/%s",
-        options.getProjectId(), keyLocation, keyRing, keyName);
+    key =
+        String.format(
+            "projects/%s/locations/%s/keyRings/%s/cryptoKeys/%s",
+            options.getProjectId(), keyLocation, keyRing, keyName);
+
+    /*
+     * Delete stale instances that have been created earlier by this test but not deleted.
+     * Backups needed to be deleted from the instance first, as the instance can only be
+     * deleted once all backups have been deleted.
+     * */
+    deleteStaleTestEncryptedInstances();
+  }
+
+  /**
+   * Deleting all the test instances with name beginning from 'encrypted-test-' and were created
+   * before 24 hours.
+   *
+   * @throws InterruptedException If Thread.sleep() interrupted
+   */
+  static void deleteStaleTestEncryptedInstances() throws InterruptedException {
+    List<Instance> instances =
+        Lists.newArrayList(
+            spanner
+                .getInstanceAdminClient()
+                .listInstances(Options.filter("name:encrypted-test-"))
+                .iterateAll());
+
+    Timestamp now = Timestamp.now();
+
+    for (Instance instance : instances) {
+      if (TimeUnit.HOURS.convert(
+              now.getSeconds() - instance.getCreateTime().getSeconds(), TimeUnit.SECONDS)
+          > 24) {
```

Comment:
Converting from a smaller unit(SECONDS) to a larger unit(HOUR), will lead to loss of precision. For example, 3600 to 7199 SECONDS will all return 1 HOUR.
All instances that got created between -24th hour to -25th hour from now will not get cleaned up here.

We can do either of the below:
- Change `> 24` to `>= 24`.
- 
```suggestion
      if (now.getSeconds() - instance.getCreateTime().getSeconds()
          > TimeUnit.SECONDS.convert(24L, TimeUnit.HOURS)) {
```
I think the second option would be better as we are not losing any precision whatsoever. We could have the threshold time `TimeUnit.SECONDS.convert(24L, TimeUnit.HOURS))` as a constant at the start of the function.

Reply:
We can better define a long final constant at class level for SECONDS_IN_ONE_DAY and use it.

` private static final long SECONDS_IN_ONE_DAY = 86400;`

Reply:
IMO, having `TimeUnit.SECONDS.convert(24L, TimeUnit.HOURS))` as a constant will be better as it is more flexible to make changes to it in the longer term. Say in the future, we had to change the number of hours (to say, 12 hours), we would have to make only a small change (`24L` -> `12L`).

Also, I think converting from a larger unit(HOURS) to a smaller unit(SECONDS) won't be a problem as we won't be at a risk of overflow here (as the number of seconds in our use case won't exceed the range of long).

Reply:
Converting from larger unit to smaller unit won't be a problem for sure, but I think defining primitive constants will avoid unnecessary calculation on the fly but as it's only limited to Integration tests and if readability is getting affected we can also define it using `TimeUnit`.

Reply:
I agree that using `TimeUnit.convert` can lead to some unnecessary calculation on the fly, but as these are integration tests and this threshold for cleanup is prone to changes with time, having `TimeUnit.convert` will give us more flexibility in terms of future changes.

Thanks for having this discussion offline as well.